src/extra/gfx/texture.cpp

```diff
@@ -166,11 +164,91 @@ struct TextureShard {
 
     // Copy the data since we can't keep a reference to the image variable
     ImmutableSharedBuffer isb(image.data, imageSize);
-    texture->init(format, int2(image.width, image.height), SamplerState(), std::move(isb));
+    texture->init(TextureDesc{.format = format, .resolution = int2(image.width, image.height), .data = std::move(isb)});
+
+    return textureVar;
+  }
+};
+
+struct RenderTargetShard {
+  static SHTypesInfo inputTypes() { return CoreInfo::AnyType; }
+  static SHTypesInfo outputTypes() { return Types::RenderTarget; }
+  static SHOptionalString help() {
+    return SHCCSTR("Defines a new default render target with \"main\" (color) and \"depth\" attachments");
+  }
+
+  SHRenderTarget renderTarget;
+  OwnedVar renderTargetVar{};
+
+  RenderTargetShard() {}
+
+  PARAM_IMPL(RenderTargetShard);
+
+  void warmup(SHContext *context) { PARAM_WARMUP(context); }
+  void cleanup() { PARAM_CLEANUP(); }
+
+  SHVar activate(SHContext *shContext, const SHVar &input) {
+    bool isInitialized = renderTargetVar.valueType == SHType::Object;
+    if (!isInitialized) {
+      static std::atomic<uint32_t> counter;
+      std::string id = fmt::format("shardsRenderTarget{}", counter++);
+      auto &vt = renderTarget.renderTarget = std::make_shared<RenderTarget>(id.c_str());
+      vt->configure("main", WGPUTextureFormat_RGBA8Unorm);
+      vt->configure("depth", WGPUTextureFormat_Depth32Float);
+
+      renderTargetVar = Var::Object(&renderTarget, gfx::VendorId, Types::RenderTargetTypeId);
+    }
+
+    return renderTargetVar;
+  }
+};
+
+struct RenderTargetTextureShard {
+  static SHTypesInfo inputTypes() { return Types::RenderTarget; }
+  static SHTypesInfo outputTypes() { return Types::Texture; }
+  static SHOptionalString help() { return SHCCSTR("Retrieve a named attachment from a render target"); }
```

Comment:
named attachment so far are hardcoded main and depth no?

Reply:
Yeah in this PR they are. In the other PR #436 you can define whatever you want, although there's still a hidden contract that they're called "color" and "depth" so passes can communicate